docassemble/AssemblyLine/data/questions/al_saved_sessions.yml

```diff
@@ -0,0 +1,264 @@
+    'al_sessions_url_ask_snapshot',
+    'al_sessions_url_ask_fast_forward',
+    # Maybe we will want these to be optional? Depends on how good review screen is.
+    # 'docket_number',
+    # 'docket_numbers',
+    #'user_ask_role',
+    #'user_role',    
+    #'user_started_case',
+  ]
+---
+# For save snapshot
+code: |
+  filtered_vars_tmp = all_variables(simplify=False)
+  simple_filtered_vars_tmp = all_variables()
+  
+  for var in al_sessions_variables_to_remove:
+    filtered_vars_tmp.pop(var, None)
+    simple_filtered_vars_tmp.pop(var, None)
+  
+  # Make all top-level DAFiles and DAFileCollections public
+  # TODO: is it worth making this recursive and traversing all attributes?
+  for var in filtered_vars_tmp:
+    if isinstance(var, DAFile) or isinstance(var, DAFileCollection) or isinstance(var, DAFileList):
+      try:
+        var.set_attributes(private=False)
+      except:
+        pass
+  
+  al_simple_filtered_vars = simple_filtered_vars_tmp
+  al_sessions_filtered_vars = filtered_vars_tmp
+  del filtered_vars_tmp
+  del simple_filtered_vars_tmp
+---
+# For fast forward
+---
+code: |
+  # vars = get_session_variables(filename, session_id, secret, simplify=False)
+  al_sessions_source_session = int(al_sessions_source_session)
+  filtered_vars_tmp = get_session_variables(
+    al_sessions_list[al_sessions_source_session].get('filename'),
+    al_sessions_list[al_sessions_source_session].get('session'),
+    simplify=False)
+   
+  for var in al_sessions_variables_to_remove:
+    filtered_vars_tmp.pop(var, None)
+
+  # Mark all top-level files public so the new session can open them
+  # TODO: is it worth making this recursive and traversing all attributes?  
+  for var in filtered_vars_tmp:
+    if isinstance(var, DAFile) or isinstance(var, DAFileCollection) or isinstance(var, DAFileList):
+      try:
+        var.set_attributes(private=False)
+      except:
+        pass
+        
+  al_sessions_fast_forward_filtered_vars = filtered_vars_tmp
+  del filtered_vars_tmp
+---
+# Triggered by "save snapshot"
+need:
+  - al_sessions_filtered_vars
+  - al_sessions_snapshot_label
+code: |
+  al_sessions_destination_session = f"{user_info().current_package}:al_saved_sessions_store.yml"
+  if not defined("al_sessions_new_session_id"):
+    al_sessions_new_session_id = create_session(al_sessions_destination_session)
+  al_sessions_filtered_vars["_internal['subtitle']"] = al_sessions_snapshot_label
+  try:
+    set_session_variables(al_sessions_destination_session, al_sessions_new_session_id, al_sessions_filtered_vars)
+    al_sessions_save_session_snapshot_success = True
+  except:
+    al_sessions_save_session_snapshot_success = False
+  al_sessions_save_session_snapshot = True  
+---
+# Triggered by "Save snapshot"
+continue button field: al_sessions_snapshot_success
+question: |
+  % if al_sessions_save_session_snapshot_success:
+  Your snapshot was saved
+  % else:
+  Something went wrong saving your snapshot
```

Comment:
The info about the failure is never saved (from the Except on like 229). Should it be logged, or just shown on this screen?